tests/plugins/rwahsNavigation/RwahsNavigationPluginConfigurationTest.php

```diff
@@ -0,0 +1,101 @@
+ *
+ * @package CollectiveAccess
+ * @subpackage tests
+ * @license http://www.gnu.org/copyleft/gpl.html GNU Public License version 3
+ *
+ * ----------------------------------------------------------------------
+ */
+
+require_once __CA_APP_DIR__ . '/plugins/rwahsNavigation/rwahsNavigationPlugin.php';
+
+/**
+ * Tests the configuration-checking functionality of the plugin.  Simply constructs the plugin with different (both
+ * valid and invalid) configuration files, and asserts against the result of calling checkStatus().
+ */
+class RwahsNavigationConfigurationPluginTest extends PHPUnit_Framework_TestCase {
+
+	public function testDefaultConfigurationIsDisabledAndValid() {
+		$vo_plugin = new rwahsNavigationPlugin(__CA_APP_DIR__ . '/plugins/rwahsNavigation');
+		$va_plugin_status = $vo_plugin->checkStatus();
+		$this->assertFalse($va_plugin_status['available'], 'The plugin is disabled by default');
+		$this->assertEmpty($va_plugin_status['errors'], 'The default configuration does not produce any errors');
+	}
+
+	public function testEnabledConfigurationWithNoShortcutsIsEnabledAndValid() {
+		$vo_plugin = new rwahsNavigationPlugin(__DIR__ . '/conf/enabled-only');
+		$va_plugin_status = $vo_plugin->checkStatus();
+		$this->assertTrue($va_plugin_status['available'], 'The plugin can be enabled by configuration');
+		$this->assertEmpty($va_plugin_status['errors'], 'Enabling the plugin without specifying search menu shortcuts does not produce any errors');
+	}
+
+	public function testCorrectMissingSearchMenuShortcutPropertiesGenerateErrors() {
+		$vo_plugin = new rwahsNavigationPlugin(__DIR__ . '/conf/missing-search-menu-properties');
+		$va_plugin_status = $vo_plugin->checkStatus();
+		$this->assertEquals(3, sizeof($va_plugin_status['errors']), 'There are three required properties');
+		$this->assertNotFalse(
+			array_search(
+				_t('Custom search shortcut with key "%1" does not specify a "%2" value, which is required', 'type1', 'type_code'),
+				$va_plugin_status['errors']
+			),
+			'The "type_code" setting is required for search menu shortcuts'
+		);
+		$this->assertNotFalse(
+			array_search(
+				_t('Custom search shortcut with key "%1" does not specify a "%2" value, which is required', 'type2', 'form_code'),
+				$va_plugin_status['errors']
+			),
+			'The "form_code" setting is required for search menu shortcuts'
+		);
+		$this->assertNotFalse(
+			array_search(
+				_t('Custom search shortcut with key "%1" does not specify a "%2" value, which is required', 'type3', 'display_code'),
+				$va_plugin_status['errors']
+			),
+			'The "display_code" setting is required for search menu shortcuts'
+		);
+	}
+
+	public function testCorrectMissingNewMenuShortcutPropertiesGenerateErrors() {
+		$vo_plugin = new rwahsNavigationPlugin(__DIR__ . '/conf/invalid-new-menu-properties');
+		$va_plugin_status = $vo_plugin->checkStatus();
+		$this->assertEquals(2, sizeof($va_plugin_status['errors']), 'There is one error in the configuration file');
```

Comment:
can `sizeof($va_plugin_status['errors'])` only equal 1 or 2? If not then the description of this error is incorrect.